src/api/mod.rs

```diff
@@ -1,27 +1,41 @@
+pub use self::cdtime::{nanos_to_collectd, CdTime};
+pub use self::logger::{collectd_log, log_err, CollectdLoggerBuilder, LogLevel};
+pub use self::oconfig::{ConfigItem, ConfigValue};
 use crate::bindings::{
-    data_set_t, hostname_g, plugin_dispatch_values, uc_get_rate, value_list_t, value_t, ARR_LENGTH,
-    DS_TYPE_ABSOLUTE, DS_TYPE_COUNTER, DS_TYPE_DERIVE, DS_TYPE_GAUGE,
+    data_set_t, hostname_g, meta_data_add_boolean, meta_data_add_double, meta_data_add_signed_int,
+    meta_data_add_string, meta_data_add_unsigned_int, meta_data_create, meta_data_t, meta_data_toc, meta_data_get_boolean, meta_data_get_double, meta_data_get_signed_int,
+    meta_data_get_string, meta_data_get_unsigned_int,
+    meta_data_type, plugin_dispatch_values, uc_get_rate, value_list_t, value_t, ARR_LENGTH,
+    DS_TYPE_ABSOLUTE, DS_TYPE_COUNTER, DS_TYPE_DERIVE, DS_TYPE_GAUGE, MD_TYPE_BOOLEAN,
+    MD_TYPE_DOUBLE, MD_TYPE_SIGNED_INT, MD_TYPE_STRING, MD_TYPE_UNSIGNED_INT,
 };
 use crate::errors::{ArrayError, CacheRateError, ReceiveError, SubmitError};
 use chrono::prelude::*;
 use chrono::Duration;
 use memchr::memchr;
+use libc;
 use std::borrow::Cow;
-use std::ffi::CStr;
+use std::collections::HashMap;
+use std::ffi::{CStr, CString};
 use std::fmt;
-use std::os::raw::c_char;
+use std::os::raw::{c_char, c_void};
 use std::ptr;
 use std::slice;
 use std::str::Utf8Error;
 
-pub use self::cdtime::{nanos_to_collectd, CdTime};
-pub use self::logger::{collectd_log, log_err, CollectdLoggerBuilder, LogLevel};
-pub use self::oconfig::{ConfigItem, ConfigValue};
-
 mod cdtime;
 mod logger;
 mod oconfig;
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum MetaValue {
```

Comment:
This enum should have a doc comment (you can preview docs with `cargo doc` and viewing the `./target/doc` folder)